addons/website_sale/controllers/main.py

```diff
@@ -748,6 +748,7 @@ def cart(self, access_token=None, revive='', **post):
             elif abandoned_order.id != request.session.get('sale_order_id'):  # abandoned cart found, user have to choose what to do
                 values.update({'access_token': abandoned_order.access_token})
 
+        order.write({'carrier_id': False})
```

Comment:
I initially did not reset the `carrier_id`. The first time I visited the cart, the delivery charge was successfully hidden. Then, I proceeded to checkout (but did not check out completely. I mean I didn't pay). After that, I returned to the /shop and then to the cart, and this time the delivery charge was not hidden.

Reply:
the delivery method is set in /checkout and it is normal that it is set before you pay in /payment :)

Reply:
[Video](https://www.awesomescreenshot.com/video/31722964?key=2f3d37d36a6d08a62e7ce095632faf53). This is how it behaves when I remove `order.write({'carrier_id': False})`. Is that okay?

Reply:
The first time we visit the `shop/cart/` page, there is no `Delivery charge` (because we get an empty list in `website_sale_order.carrier_id`. Now when we again go to `shop/cart/` after going through `/payment` or `/checkout`, we are able to see the Delivery charge because this time `website_sale_order.carrier_id` is not an empty list.

Reply:
of course, it is the point of the task, if the delivery method is set, then you need to see the price in the cart summary.
![image](https://github.com/user-attachments/assets/147f6eaf-0e00-4be4-bfb1-d64c9b4f173d)

Reply:
Okay, if that's the intended behavior, I will remove that line from the controller. I was trying to replicate the behavior of 17.0.

Reply:
Great, the behavior was changed for a reason :)